identity-infrastructure/src/main/java/life/qbic/identity/infrastructure/UserJpaRepository.java

```diff
@@ -60,13 +61,62 @@ public Optional<User> findUserByUserName(String userName) {
   }
 
   @Override
-  public List<User> findByUserNameContainingIgnoreCaseAndActiveTrue(String userName,
-      Pageable pageable) {
-    return userRepo.findAllByUserNameContainingIgnoreCaseAndActiveTrue(userName, pageable);
+  public List<User> queryActiveUsersWithFilter(String filter, Pageable pageable) {
+    Specification<User> userSpecification = generateUserFilterSpecification(filter);
+    return userRepo.findAll(userSpecification, pageable).getContent();
   }
 
   @Override
   public Optional<User> findByOidcIdEqualsAndOidcIssuerEquals(String oidcId, String oidcIssuer) {
     return userRepo.findByOidcIdEqualsAndOidcIssuerEquals(oidcId, oidcIssuer);
   }
+
+  private Specification<User> generateUserFilterSpecification(String filter) {
+    Specification<User> isBlankSpec = UserSpec.isBlank(filter);
+    Specification<User> isFullName = UserSpec.isFullName(filter);
+    Specification<User> isUserNameSpec = UserSpec.isUserName(filter);
+    Specification<User> isOrcidSpec = UserSpec.isOrcid(filter);
+    Specification<User> isActiveSpec = UserSpec.isActive();
+    Specification<User> filterSpecification =
+        Specification.anyOf(isFullName,
+            isUserNameSpec,
+            isOrcidSpec
+        );
+    return Specification.where(isBlankSpec)
+        .and(filterSpecification)
+        .and(isActiveSpec);
+  }
+
+  private static class UserSpec {
+
+    //If no filter was provided return all Users
+    public static Specification<User> isBlank(String filter) {
+      return (root, query, builder) -> {
+        if (filter != null && filter.isBlank()) {
+          return builder.conjunction();
+        }
+        return null;
+      };
+    }
+
+    public static Specification<User> isUserName(String filter) {
+      return (root, query, builder) ->
+          builder.like(root.get("userName"), "%" + filter + "%");
+    }
+
+    public static Specification<User> isFullName(String filter) {
+      return (root, query, builder) ->
+          builder.like(root.get("fullName"), "%" + filter + "%");
+    }
+
+    public static Specification<User> isOrcid(String filter) {
+      return (root, query, builder) ->
+          builder.like(root.get("oidcId"), "%" + filter + "%");
+    }
```

Comment:
careful. OpenId Connect (oidc) is not equal to ORCiD!
You might need to filter for the issuer. For example, "tester von und zu QBiC" does not have an ORCiD but uses only their sandbox instance.
You could introduce a property in the applicaiton properties specifying the issuer Url(s) to consider here.

Reply:
True, I renamed the specification and added a comment that it should be extended if multiple oidc are employed for a user. Currently we only want to check if a provided string contains any oidc value